### PR TITLE
wlroots: update to 0.10.0, cage: revbump, wayfire: update to 0.3.1, sway: update to 1.4

### PR DESCRIPTION
**Pull request by damneq**

### CODE REVIEW

**Comment by xtraeme** [comment]
soname bump?

**Comment by damneq** [comment]
Yes, anything else I need to do when that happens?

**Comment by damneq** [comment]
Revbumped all packages that xbps told me to when compiling with the old soname

**Comment by fosslinux** [comment]
Please use the proper commit message format, eg

`cage: bump for wlroots update` or something along those lines.

See https://github.com/void-linux/void-packages/blob/master/Manual.md#contributing-via-git

**Comment by fosslinux** [comment]
Also seems like the revbump for wayfire broke it... you'll need to look into that.

**Comment by damneq** [comment]
Sorry about not using the proper commit message format.
As for breaking stuff it broke sway and cage too. I managed to compile sway-1.3-rc3 but since that's an rc I can't update it yet, I think it comes down to packages adding compatibility for newer versions of wlroots so nothing I can do right now. Should I close this pull request and make another one when new versions of mentioned packages will be released or just wait and update this pull request?

**Comment by damneq** [comment]
Okay, so I did my research, compiled everything a bunch of times and I can confirm that we have to wait for new versions of sway and cage. A new version of wayfire has been already released and it compiles fine with wlroots 0.9.1, the question is can I update it in this pull request or do I have to make another one?

**Comment by fosslinux** [comment]
Yep, do the ones you can. Leave it open so that people know it's being worked on while you wait.

However there are some patches already avaliable in master which you can add to the patches/ subdirectory for each package:
- Sway: https://github.com/swaywm/sway/commit/ed02261551ac7748d38af4a0bcdeeab0437571fe -- adding .patch to the end gives you a patch you can use https://github.com/swaywm/sway/commit/ed02261551ac7748d38af4a0bcdeeab0437571fe.patch
- Cage: https://github.com/Hjdskes/cage/commit/51e6c760da51e2b885737d61a61cdc965bb9269d and https://github.com/Hjdskes/cage/commit/bfd7d605b89fc5436505611ad01806fc11aa9a08.

For future reference, I found these by going to the github repositories and searching for wlroots in commits.

**Comment by damneq** [comment]
Thank you for the patches, but it's not as easy as it seems. I had to add new packages to makedepends and also use a few more patches from upstream but I managed to compile cage using wlroots 0.9.1, I'll test compiling sway next and when everything will be good I will push the changes here.

**Comment by damneq** [comment]
As it turns out making sway-1.2 work (compile) with wlroots-0.9.1 is a lot more compilacted because a lot of files were changed a lot (and I really mean it) before updating it to work with newer versions of wlroots. Patching it would require me to make a frankenstein mix between sway-1.2 and sway-1.3 and I don't think that's a good option. All we need is to wait for sway-1.3 to be released and then it will work out of the box (well kinda, new makedends are required but it's not a big problem). For now I can update this pull request with new version of wayfire and patches for cage, then we have to wait for sway-1.3.

**Comment by damneq** [comment]
Sorry for the mess in the commits, I'll squash all of them into one when sway-1.3 will come out.

**Comment by damneq** [comment]
I redid all the comits to delete that empty one.
New version of sway was released but it changed the meson.build to require wlroots>0.10.0 so I patched it to work with 0.9.1 as 0.10.0 isn't released as of yet. I'd really appreciate if someone would go over this and see if everything is ok.

**Comment by Wychmire** [comment]
wlroots 0.10.0 [has a release now](https://github.com/swaywm/wlroots/releases/tag/0.10.0) so you should be able to remove the patch.

Since you're updating sway in this PR as well, you should probably update swayidle to [1.6](https://github.com/swaywm/swayidle/releases/tag/1.6) and swaylock to [1.5](https://github.com/swaywm/swaylock/releases/tag/1.5).

**Comment by damneq** [comment]
Well it's a pre-release so I didn't know if I could use it or not. I'll update wlroots, swaylock and swayidle then.

**Comment by Wychmire** [comment]
oh, sorry. Didn't check to see if it was a pre-release or not, just that it existed. Should probably stick with 0.9.1 and the patch for now I think

**Comment by damneq** [comment]
Ok, just swaylock and swayidle will be updated then.

File: srcpkgs/cage/patches/cage-0.1.1-fix-building-with-wlroots-0.9.1.patch [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ 	/* Place the cursor in the center of the screen. */
+ 	wlr_cursor_warp(server->seat->cursor, NULL, wlr_output->width / 2, wlr_output->height / 2);
+ 	wlr_output_damage_add_whole(server->output->damage);
++
++	wlr_output_enable(wlr_output, true);
++	wlr_output_commit(wlr_output);
+ }
+ 
+ void
+--- .builds/alpine.yml
```
Comment (fosslinux) [inline srcpkgs/cage/patches/cage-0.1.1-fix-building-with-wlroots-0.9.1.patch]: You can probably remove everything from this line

File: srcpkgs/cage/patches/cage-0.1.1-fix-building-with-wlroots-0.9.1.patch [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+@@ -18,8 +18,7 @@
+   # version, instead of master, to avoid any breaking changes in wlroots.
+   - wlroots: |
+       cd wlroots
+-      # This corresponds to the tag of 0.6.0
+-      git checkout c0305f4f864543f8c3fea6f302e91c9b1d3396f3
++      git checkout 0.9.1
+       meson --prefix=/usr/local build -Drootston=false -Dexamples=false
+       ninja -C build
+       sudo ninja -C build install
```
Comment (fosslinux) [inline srcpkgs/cage/patches/cage-0.1.1-fix-building-with-wlroots-0.9.1.patch]: ... to this line. As this is just for the builds they use interally and dosen't impact on our builds.

Comment (damneq) [reply srcpkgs/cage/patches/cage-0.1.1-fix-building-with-wlroots-0.9.1.patch]: Right, tested it on my side and it seems to working fine, will commit in a while.

**Comment by damneq** [comment]
Revised the patch.
I guess I'll ask here, what does it take to become a maintainer of a package? wlroots is currently orphaned and I'd like to take over the maintainership, thanks in advance.

**Comment by travankor** [comment]
the freerdp stuff in `makedepends` is wrong. freerdp support is removed from wlroots (0.10.0).
swaylock and swayidle do not depend on wlroots, so you don't need to update them in this PR.

pretty much anyone can be maintainer of a package, but the actual "maintainership" of packages is loosely held in Void

**Comment by damneq** [comment]
Now (0.10.0) it's removed, but when bulding against wlroots 0.9.1 it would fail, will redo.

**Comment by damneq** [comment]
Everything should be good now and the pull request should be ready to merge.

**Pull request merged**